internal/app/referenceclient/impl.go

```diff
@@ -333,6 +339,36 @@ func (i *invoker) bidiStream(
 	return result, nil
 }
 
+func (i *invoker) unimplemented(
+	ctx context.Context,
+	req *v1alpha1.ClientCompatRequest,
+) (*v1alpha1.ClientResponseResult, error) {
+	msg := req.RequestMessages[0]
+	ur := &v1alpha1.UnimplementedRequest{}
+	if err := msg.UnmarshalTo(ur); err != nil {
+		return nil, err
+	}
+
+	request := connect.NewRequest(ur)
+
+	var protoErr *v1alpha1.Error
+
+	// Invoke the Unary call
+	_, err := i.client.Unimplemented(ctx, request)
+	if err != nil {
```

Comment:
This seems a little odd that this doesn't do the same thing as `Unary`. In fact, you could make that method generic (parameterized by the request/response types), so the exact same function could be used for _all_ unary RPCs (admittedly, there are only two of them).

Reply:
I don't think you can make a struct method genericized, right? I tried to implement what you mentioned and kept running into issues (unless I'm missing an easiser way).

Reply:
Indeed, it can't be a method. It has to be a function. So you'd have to pass `i` as a normal parameter instead of as a method receiver.

In any event, this was a non-blocking comment. I guess a simpler option is to just return an error instead of an RPC response if `err == nil`.

Reply:
I think we can get even simpler than that. The `ConvertErrorToProtoError` function just returns `nil` if `err == nil`, so we can just set whatever that returns on the `Error` property and return. The test will fail if the error is nil in that case then.